main/resource-access.bs

```diff
@@ -142,19 +139,151 @@ through POST, using the HTTP `Slug` header as defined in [[!RFC5023]].
 Clients who want the server to determine the URI for their resource, MUST use
 the POST request.
 
-Servers MUST allow creating new resources with a POST request to URI path
+[[Source](https://github.com/solid/specification/pull/160#issuecomment-636822687)].
+
+### Reading resources ### {#read}
+
+Servers MUST support the HTTP `GET`, `HEAD` and `OPTIONS` methods [[!RFC7231]]
+for clients to read resources or to determine communication options.
+[[Source](https://github.com/solid/specification/issues/39#issuecomment-538017667)]
+
+When using Web Access Control ([[#wac]]):
```

Comment:
Why is this relevant?

Reply:
Does https://github.com/solid/specification/pull/188/commits/5ec6ad6536d5fbb4361399f432c7b7d67a6ec665 also address this?

Reply:
No, I'm not sure how this relates to WAC; value of the `Allow` header(s) should be independent of current user, right?

Reply:
I'm not sure what would determine the allowed methods if not derived from an access policy. Alternatively, we'd expect all methods; GET, HEAD, OPTIONS, PUT, POST, PATCH, DELETE for every URL?

(Putting aside the other issue on if a server can be read-only.)

Reply:
> I'm not sure what would determine the allowed methods

Server support (so read-only or not is not a side issue for me here).

Reply:
I said aside for the moment because I didn't want to write out two sets above. So either:
* GET, HEAD, OPTIONS
* GET, HEAD, OPTIONS, PUT, POST, PATCH, DELETE

I suppose it is also possible to have a server that only allows creating, updating, deleting:
* PUT, POST, PATCH, DELETE

Noting that if only general server support (=what the spec prescribes) is listed, then we need something like WAC-Allow in order to help the client for their next request. It still doesn't tell them which methods are allowed - so potentially making the request but no success.

Noting also that having the server in read-only mode entails that information is coming from somewhere but presumably not the authorization policy. Otherwise, might as well map the access modes to the methods and list them in Allow.

Reply:
> then we need something like WAC-Allow in order to help the client for their next request

Yeah (but preferably a better mechanism).

But that is my reading of https://tools.ietf.org/html/rfc7231#section-7.4.1, which states:

> The "Allow" header field lists the set of methods advertised as supported by the target resource.

Note that it is about the _resource_, not about the current user-agent.

Although below, it says:

> The actual set of allowed methods is defined by the origin server at the time of each request.

And then again:

> An empty Allow field value indicates that the resource allows no methods

So I still see my interpretation validated (but the evidence is not explicit).

> Noting also that having the server in read-only mode entails that information is coming from somewhere

It might literally be a temperature sensor (and IoT was actually the case I was thinking of), with a specific ACL for reads still.

The difference is that, with a read/write server, the ACL might change. So `POST` is _supported_, just not _permitted_ for a certain user.

Reply:
Henry captured the considerations well here:
https://github.com/solid/solid/wiki/Client-Authentication-Discovery#allow-headers

I'm fine with read-only or write-only use cases.

I'm not sure what you mean exactly re "specific ACL for reads still". Just want to understand if there is a way beyond an implementation configuration. In that case, we can qualify the operations eg. "Servers that want to enable read operations:" and similarly for the other operations.

Not specifying where the methods are coming from in addition to what the RFC says is also an option.

Reply:
> I'm not sure what you mean exactly re "specific ACL for reads still".

That was in response to you writing

> Otherwise, might as well map the access modes to the methods and list them in Allow.

where I wanted to remark that even in a read-only server, not all users might have read permissions, so such a mapping wouldn’t necessarily exist. So read-only and still having auth aren’t mutually exclusive.

Reply:
So would removing "When using Web Access Control" also cover the read-only and write-only cases? Essentially how the methods in Allow are determined becomes an implementation detail.

Reply:
Works for me!

Reply:
https://github.com/solid/specification/pull/188/commits/51bbd733b2feaf2073ab6d8bf720332c67f9fb0c